cns/service/main.go

```diff
@@ -651,17 +651,23 @@ func main() {
 		return
 	}
 
-	if cnsconfig.ChannelMode == cns.Managed {
+	// copy ChannelMode from cnsconfig to HTTPRemoteRestService config
+	switch cnsconfig.ChannelMode {
+	case cns.Direct:
+		config.ChannelMode = cns.Direct
+	case cns.Managed:
 		config.ChannelMode = cns.Managed
 		privateEndpoint = cnsconfig.ManagedSettings.PrivateEndpoint
 		infravnet = cnsconfig.ManagedSettings.InfrastructureNetworkID
 		nodeID = cnsconfig.ManagedSettings.NodeID
-	} else if cnsconfig.ChannelMode == cns.CRD {
+	case cns.CRD:
 		config.ChannelMode = cns.CRD
-	} else if cnsconfig.ChannelMode == cns.MultiTenantCRD {
+	case cns.MultiTenantCRD:
 		config.ChannelMode = cns.MultiTenantCRD
-	} else if acn.GetArg(acn.OptManaged).(bool) {
-		config.ChannelMode = cns.Managed
+	default:
+		if acn.GetArg(acn.OptManaged).(bool) {
+			config.ChannelMode = cns.Managed
+		}
```

Comment:
in every mode except `cns.Managed`, we're just copying the mode, right?
it seems like we could remove...most of this:
```go
	// copy ChannelMode from cnsconfig to HTTPRemoteRestService config
	config.ChannelMode = cnsconfig.ChannelMode
	if config.ChannelMode == cns.Managed {
		privateEndpoint = cnsconfig.ManagedSettings.PrivateEndpoint
		infravnet = cnsconfig.ManagedSettings.InfrastructureNetworkID
		nodeID = cnsconfig.ManagedSettings.NodeID
	}
	if acn.GetArg(acn.OptManaged).(bool) {
		config.ChannelMode = cns.Managed
	}
```

Reply:
refactored, it appears this should be fine